webpack.config.ts

```diff
@@ -0,0 +1,367 @@
+        let firefoxVariant: 'fennec' | 'geckoview' | undefined = undefined
+        let architecture: 'web' | 'app' = 'web'
+        let resolution: 'desktop' | 'mobile' = 'desktop'
+        let buildType: 'stable' | 'beta' | 'insider' = 'stable'
+        if (target.Chromium) buildTarget = 'chromium'
+        if (target.Firefox) buildTarget = 'firefox'
+        if (target.FirefoxForAndroid) firefoxVariant = 'fennec'
+        if (target.StandaloneGeckoView) {
+            firefoxVariant = 'geckoview'
+            architecture = 'app'
+        }
+        if (target.Safari) {
+            buildTarget = 'safari'
+            architecture = 'app'
+        }
+        if (architecture === 'app' || firefoxVariant === 'fennec') resolution = 'mobile'
+        if (target.E2E) buildTarget = 'E2E'
+        if (target.Beta) buildType = 'beta'
+        if (target.Insider) buildType = 'insider'
+
+        // build the envs
+        const allEnv = {
+            STORYBOOK: false,
+            target: buildTarget,
+            build: buildType,
+            architecture,
+            resolution,
+        }
+        if (firefoxVariant) allEnv[firefoxVariant] = firefoxVariant
+        return allEnv
+    }
+    function getSSRPlugin() {
+        if (env === 'development') return []
+        return [new SSRPlugin('popup.html', src('./src/extension/popup-page/index.tsx'))]
+    }
+}
+
+/** All targets available: --firefox --firefox=nightly --firefox-android --firefox-gecko --chromium --wk-webview --e2e */
+function getBuildPresets(argv: any) {
+    return {
+        Firefox: (argv.firefox || argv['firefox-android'] || argv['firefox-gecko']) as 'nightly' | boolean,
+        FirefoxDesktop: argv.firefox as string | boolean,
+        FirefoxForAndroid: !!argv['firefox-android'],
+        StandaloneGeckoView: !!argv['firefox-gecko'],
+        Chromium: !!argv.chromium,
+        Safari: !!argv['wk-webview'],
+        E2E: !!argv.e2e,
+        Beta: !!argv.beta,
+        Insider: !!argv.insider,
+        ReproducibleBuild: !!argv['reproducible-build'],
+    }
+}
+/** Get git info */
+function getGitInfo() {
+    if (git.isRepository())
+        return {
+            BUILD_DATE: new Date().toISOString(),
+            VERSION: git.describe('--dirty'),
+            TAG_NAME: git.tag(),
+            COMMIT_HASH: git.commitHash(),
+            COMMIT_DATE: git.commitDate().toISOString(),
+            REMOTE_URL: git.remoteURL(),
+            BRANCH_NAME: git.branchName(),
+            DIRTY: git.isDirty(),
+            TAG_DIRTY: git.isTagDirty(),
+        }
+    return {
+        BUILD_DATE: new Date(0).toISOString(),
+        VERSION: require('./package.json').version + '-reproducible',
+        TAG_NAME: 'N/A',
+        COMMIT_HASH: 'N/A',
+        COMMIT_DATE: 'N/A',
+        REMOTE_URL: 'N/A',
+        BRANCH_NAME: 'N/A',
+        DIRTY: false,
+        TAG_DIRTY: false,
+    }
+}
+
+function toArray(x: string | string[]) {
+    return typeof x === 'string' ? [x] : x
+}
```

Comment:
```suggestion
function toArray(x: string | string[]) {
    return typeof x === 'string' ? [x] : x
}
```

move to https://github.com/DimensionDev/kit ?

Reply:
nope